docs/content/tutorial/ingress-tutorial.md

```diff
@@ -0,0 +1,139 @@
+title: "Exposing functions with Ingress"
+draft: false
+weight: 43
+---
+
+This tutorial will walk you through exposing a function using a ingress controller. We will make the function available on a FQDN using Fission's route and ingress controller setup in a cloud environment.
+
+## Setup & pre-requisites
+
+You will need a Kubernetes cluster with Fission installed (Please check [installation page](../../installation/installation/) for details). This tutorial uses a cloud load balancer and if you are trying on Minikube, you might want [to take a look at details here](https://github.com/kubernetes/minikube/issues/496)
+
+Later parts of this tutorial use a FQDN to reach the function. If you plan to go along in this section, you will need a domain name setup and access to modify the NS records & create A record in the zone of the domain name you have. The tutorial uses Google cloud to walk through the tutorial but you can use any cloud you prefer to. Also the changes in name server can take upto 24-48 hours so you may want to use an already created domain name.
+
+### Setup an Ingress Controller
+
+First thing we will need is an ingress controller and we will use Nginx ingress controller in this tutorial. Based on your setup you can choose [one of the ways to install Nginx ingress controller from here](https://kubernetes.github.io/ingress-nginx/deploy/). This setup should work with other ingress controllers also but has not been tested.
+
+Once ingress controller is installed properly, let's verify that the installation succeeded (In my case all components of ingress controller were installed in ingress-nginx namespace):
+
+```
+kubectl get all -n ingress-nginx
+NAME                                           READY     STATUS    RESTARTS   AGE
+po/default-http-backend-66b447d9cf-4q8f7       1/1       Running   0          19d
+po/nginx-ingress-controller-58fcfdc6fd-2cwts   1/1       Running   0          19d
+
+NAME                       CLUSTER-IP      EXTERNAL-IP      PORT(S)                      AGE
+svc/default-http-backend   10.11.243.109   <none>           80/TCP                       19d
+svc/ingress-nginx          10.11.245.254   35.200.150.175   80:31000/TCP,443:30666/TCP   19d
+
+NAME                              DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+deploy/default-http-backend       1         1         1            1           19d
+deploy/nginx-ingress-controller   1         1         1            1           19d
+
+NAME                                     DESIRED   CURRENT   READY     AGE
+rs/default-http-backend-66b447d9cf       1         1         1         19d
+rs/nginx-ingress-controller-58fcfdc6fd   1         1         1         19d
+
+```
+Some key points to validate that ingress controller installation succeeded:
+
+- The ingress controller pod is up and running
+- The ingress-nginx service has a external IP address populated
+- If you hit the external IP address of the ingress-nginx, you get the default backend page:
```

Comment:
You could add some example commands for each of the pointers (in parentheses). For the first two ones.

Reply:
Since commands & names might vary, will point to output from previous command instead of showing commands here.